Blackjack.py

```diff
@@ -0,0 +1,282 @@
+                for rank_card in ranks:
+                    self.all_cards.append(Card(suits_card, rank_card))
+
+        def shuffle(self):
+            random.shuffle(self.all_cards)
+
+        def deal(self):
+            return self.all_cards.pop()
+
+        def __str__(self):
+            for x in range(0, len(self.all_cards)):
+                print(f"{self.all_cards[x]}")
+            return f"This is the current cards in the deck"
+
+
+    class Hand():
+
+        def __init__(self):
+            self.all_cards = []
+            # player overall value towards 21 = winning! Blackjack
+            self.player_value = 0
+            # counter to keep track of when an ace is in player hand
+            self.ace_counter = 0
+
+        # adding a card to player hand, calculating value towards 21 & Ace counter
+        def add_card(self, new_card):
+            self.all_cards.append(new_card)
+            self.player_value += new_card.value
+            if new_card.rank == "Ace":
+                self.ace_counter += 1
+
+        # Best way to change the Ace value is by subtracting from the overall value sum
+        def adjust_ace(self):
+            while self.player_value > 21 and self.ace_counter:
+                self.player_value -= 10
+                self.ace_counter -= 1
+
+        def __str__(self):
+            # for x in range(0,len(self.all_cards)):
+            # print(f"{self.all_cards[x]}")
+            return f"Hand Value is {self.player_value}"
+
+
+    class Chips():
+
+        def __init__(self):
+            self.total = 100  # lets say this is our starting chips
+            self.bet = 0
+
+        def win_bet(self):
+            self.total += self.bet
+
+        def lose_bet(self):
+            self.total -= self.bet
+
+
+    def take_bet(chips):
+
+        while True:
+            try:
+                chips.bet = int(input("Place your bet Player: "))
+
+            except ValueError:
+                print("You must bet an integer value")
+
+            else:
+                if chips.bet > chips.total:
+                    print("You exceeded your total of {chips.total}")
+                else:
+                    break
+
+
+    def hit(deck, hand):
+        hand.add_card(deck.deal())
+        hand.adjust_ace()
+
+
+    def hit_or_stand(deck, hand):
+        global playing
+
+        while True:
+
+            turn = input("\nWill you (hit) or (stand)? (please type one of the following under paranthesis)\n")
+
+            if turn == "hit":
+                hit(deck, hand)
+            elif turn == "stand":
+                playing = False
+            else:
+                print('input must be either "hit" or "stand"! Try again')
+                continue
+            break
+
+
+    def show_some(player, dealer):
+        print("\nShow me only some cards yo ;)")
+        print(f"\nDealer's hand:\nhidden\n{dealer.all_cards[1]}")
+        print("\nPlayer's hand:", *player.all_cards, player, sep="\n")

```

Comment:
Just want to point out and make sure you under stand what `*` is doing. It's not a "pointer" like C, but means to "unpack" the all the items in the list.
If this were a dictionary and not a list you would use `**dictionary_items` since there are two things per item.

This is important to know for function usage later on so you can send or receive arbitrary amounts are args. Through `packing` or `unpacking` we can go over specific later when you see it used or when it makes sense to use it.

Reply:
Ah thank you. Did not know that. I learned this in the course but did not know about dictionary_items.